src/circular_queue.cpp

```diff
@@ -12,8 +12,9 @@ CircularQueue::CircularQueue(int num_elements, int size_of_elm):
 
 	// malloc the memory for the circular queue
 	queue_array = (queue_elm *) malloc(sizeof(queue_elm) * len);
+	data_array = (char *) malloc(elm_size * len * sizeof(char));
```

Comment:
sorry, I meant having the data be in the same array as `queue_array`. as in we have the data in the same array as the queue metadata. what do you think about that?

Reply:
Even better. We'll need some error checking to ensure we don't overwrite it but it should be another easy change.

Reply:
I'm slightly apprehensive, because if this data structure is swapped out, we may need to do multiple IOs to access a buffer. Also I'm concerned this may do weird things for alignment

Reply:
Actually, after thinking some more this isn't so easy a change and would introduce a lot of complexity. Unless all the metadata goes at the beginning in which case I don't think this helps us anymore.

So I think we should go without it. Unless you can come up with a good way to implement.
[[meta data][data][metadata][data] ... ]

Reply:
maybe we could treat this as an OS malloc assignment and manually add metadata blocks into a big chunk of array space. but that sounds complicated so yeah we can table it